### PR TITLE
To-Do List!

### PR DESCRIPTION
**Pull request by lmaldonadoch**
On this project, we used classes to organize our app in different components. We applied all the previously acquired knowledge of modules, factories, and basic Vanilla JavaScript concepts and elements. We created an app in which the user can create new projects and tasks to complete. We added a button to marked them as completed and to undo the completion. Users can also edit and delete tasks and delete projects.

# Project Milestones
- Use Webpack introduced in the previous lesson.
- Remember to include all Webpack related packages in devDependencies.
- Remember to add scripts that are necessary to build your project in package.json.
- Remember to add node_modules dir to .gitignore.
- Use ES6 introduced in the previous lesson.

# Requirements

## Section Requirements:
- Keep your code clean.
Do not commit to your repo. It’s ugly, it kills performance and it can make confidential data be visible in the browser tools.
- Keep the number of changes/updates to the DOM as low as possible, they are very expensive for the browser.
- Keep the application logic separated from DOM manipulation tasks.
- Keep all the styling logic in CSS files.
- If you use node modules - .lock files should be in the repository to avoid problems with future versions of the dependencies.
- If you use ES6 - use object destructuring to get the values from an object. This way you can avoid repeating a lot of code.

## General requirements:
- [x] Add a descriptive README file to your repositories as described.
- [x] Use meaningful names for your Pull Requests and add a summary of its content as described.
- [x] Use descriptive commit messages as described.
- [x] Do not commit files that are not related to the project. Instead, add them to .gitignore file.
- [x] Do not commit inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history.

## Git Flow:
- [x] Use a specific branch for each feature.
- [x] Use a descriptive name for your branches.
- [x] Use Git to validate your code.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+  const saveLastProject = (project) => {
+    localStorage.setItem('lastProject', JSON.stringify(project.title));
+  };
+
+  // validates project is filled out
+
+  const validateProject = () => {
+    const form = document.getElementById('project-form');
+    if (form[0].value === '') {
+      return alert('Title must be filled out'); // eslint-disable-line no-alert
```
Comment (bellom) [inline src/main.js]: This should be production code, we shouldn't have `alert`, please remove the `alert`.

File: src/main.js [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+
+  const validateToDo = (project = null, toDo = null) => {
+    const form = document.getElementById('todo-form');
+    if (
+      form[0].value === ''
+      || form[1].value === ''
+      || form[2].value === ''
+      || form[3].value === ''
+    ) {
+      return alert('Form should be completely filled out.'); // eslint-disable-line no-alert
```
Comment (bellom) [inline src/main.js]: This should be production code, we shouldn't have `alert`, please remove the `alert`.

File: src/main.js [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+    submit.classList.add('form-button');
+    submit.onclick = () => validateToDo(project, toDo);
+    if (toDo === null) {
+      submit.innerHTML = 'Create Task';
+    } else {
+      submit.innerHTML = 'Save Changes';
+    }
+
+    return form.appendChild(submit);
+  };
```
Comment (bellom) [inline src/main.js]: Let's separate the DOM from application logic, you can create another file for DOM.

image

https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622602/

File: src/main.js [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+      window.location.reload();
+    };
+
+    const editButton = document.createElement('button');
+    editButton.innerHTML = 'Edit Task';
+    editButton.classList.add(`${toDo.title.replace(/\s/g, '')}`);
+
+    editButton.onclick = () => {
+      renderForm(project, toDo);
+    };
```
Comment (bellom) [inline src/main.js:177]: Let's prefilled the form with the data already inputted instead of being an empty new form.

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: src/main.js (hunk 1 above)
Comment (lmaldonadoch) [reply src/main.js]: Done! Thanks for pointing it out, we changed the alert for a "required" attribute for each form field

File: src/main.js (hunk 2 above)
Comment (lmaldonadoch) [reply src/main.js]: Done! Thanks for pointing it out, we changed the alert for a "required" attribute for each form field

File: src/main.js (hunk 3 above)
Comment (lmaldonadoch) [reply src/main.js]: We talked to @bellom on Slack and we agreed that this function is just creating a button in DOM and not applying any application logic.

File: src/main.js (hunk 4 above)
Comment (lmaldonadoch) [reply src/main.js:177]: Done! Thank you for pointing it out, when creating the form we just filled it out with the values of the current To Do.

**Review by bellom — APPROVED** [review]
## Hi Team :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit project completion form.
Good luck in the next project :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**